generators/utilities/src/main/resources/csrc/emulator.cc

```diff
@@ -267,25 +279,25 @@ int main(int argc, char** argv)
 
 #if VM_TRACE
   Verilated::traceEverOn(true); // Verilator must compute traced signals
+#if CY_FST_TRACE
+  std::unique_ptr<VerilatedFstC> tfp(new VerilatedFstC);
+  if (vcdfile_name) {
+    tile->trace(tfp.get(), 99);  // Trace 99 levels of hierarchy
+    tfp->open(vcdfile_name);
+  }
+#else
   std::unique_ptr<VerilatedVcdFILE> vcdfd(new VerilatedVcdFILE(vcdfile));
   std::unique_ptr<VerilatedVcdC> tfp(new VerilatedVcdC(vcdfd.get()));
   if (vcdfile) {
     tile->trace(tfp.get(), 99);  // Trace 99 levels of hierarchy
     tfp->open("");
   }
-#endif
-
-  jtag = new remote_bitbang_t(rbb_port);
-  dtm = new dtm_t(htif_argc, htif_argv);
-  tsi = new tsi_t(htif_argc, htif_argv);
+#endif // CY_FST_TRACE
+#endif // VM_TRACE
 
   signal(SIGTERM, handle_sigterm);
 
   bool dump;
-  // start reset off low so a rising edge triggers async reset
-  tile->reset = 0;
```

Comment:
I don't think your do while has the same reset behavior.
We just had an issue with this so we shouldn't lose this fix.

Reply:
yeah. john pointed that out above. i wasn't aware of that fix and made a hasty mistake.

Reply:
is there a test-case i can exercise this reset code with?

Reply:
CI should catch it deterministically, I believe. The spiflash tests fail without this, so if you want to run those on your end that should show it

Reply:
confirmed spiflash works with multithreading enabled.